### ISSUE TITLE
Small spelling improvements

### ISSUE THREAD

**Issue by PanderMusubi** [pylint-dev/pylint#8594]
### Bug description

There are some small spelling improvements that can be done realted to the use of the English article `a` or `an`. See https://en.wikipedia.org/wiki/English_articles

### Configuration

```ini
n/a
```

### Command used

```shell
n/a
```

### Pylint output

```shell
See this line https://github.com/pylint-dev/pylint/blob/main/tests/functional/n/not_callable.py#L147 for `a attribute`.

See 39 occurrences of `a f-string`.
```

### Expected behavior

Should be `an attribute` and `an f-string`

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.3
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]
```

### OS / Environment

Pop!_OS 22.04 LTS

### Additional dependencies

n/a
